### PR TITLE
ref(*): kubernetes v1.12 support

### PR DESCRIPTION
**Pull request by adamreese**

### CODE REVIEW

**Comment by adamreese** [comment]
To test in cluster run `helm init --upgrade --tiller-image quay.io/adamreese/tiller:kube-v1.12`

File: pkg/kube/client.go [hunk 1]
```diff
@@ -561,7 +561,9 @@ func (c *Client) watchUntilReady(timeout time.Duration, info *resource.Info) err
 	// In the future, we might want to add some special logic for types
 	// like Ingress, Volume, etc.
 
-	_, err = watch.Until(timeout, w, func(e watch.Event) (bool, error) {
+	ctx, cancel := watchtools.ContextWithOptionalTimeout(context.Background(), timeout)
+	defer cancel()
+	_, err = watchtools.UntilWithoutRetry(ctx, w, func(e watch.Event) (bool, error) {
```
Comment (bacongobbler) [inline pkg/kube/client.go]: would it be valuable to use `UntilWithSync` instead, or do we want to intentionally close out the connection when we hit the timeout? Based on the docstrings client-go provided, it looks like they'll [remove this endpoint eventually](https://github.com/kubernetes/client-go/blob/3e116d45271ab22b04a40641e7d431990b2349fd/tools/watch/until.go#L53-L57), and they [recommend using `UntilWithSync`](https://github.com/kubernetes/client-go/blob/3e116d45271ab22b04a40641e7d431990b2349fd/tools/watch/until.go#L108-L109).

What do you think?

File: pkg/kube/client.go [hunk 2]
```diff
@@ -654,31 +656,33 @@ func (c *Client) watchPodUntilComplete(timeout time.Duration, info *resource.Inf
 	}
 
 	c.Log("Watching pod %s for completion with timeout of %v", info.Name, timeout)
-	_, err = watch.Until(timeout, w, func(e watch.Event) (bool, error) {
+	ctx, cancel := watchtools.ContextWithOptionalTimeout(context.Background(), timeout)
+	defer cancel()
+	_, err = watchtools.UntilWithoutRetry(ctx, w, func(e watch.Event) (bool, error) {
```
Comment (bacongobbler) [inline pkg/kube/client.go]: same question here in regards to `WaitUntilSync`

**Review by bacongobbler — COMMENTED** [review]
One question about how we want to watch objects for state changes; everything else looks great!

File: pkg/kube/client.go (hunk 1 above)
Comment (adamreese) [reply pkg/kube/client.go]: That function signature is very different, I would rather use as close to a drop-in as possible for helm2

**Review by bacongobbler — APPROVED** [review]

**Pull request merged**